### ISSUE TITLE
Feature request: Support for formatting columns

### ISSUE THREAD

**Issue by systemmonkey42** [junegunn/fzf#1760]
Feature request: Support for formatting columns

- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [x] bash
    - [ ] zsh
    - [ ] fish
    
## Problem / Steps to reproduce
Can input which consists of multiple fields separated by a known separator (, or | or ; etc) be
formatted for display by converting visually to columns?

The --nth and --with-nth will continue to behave as they do, but while the input data could be
a 'CSV' (and the output of selected rows also in CSV) have the data displayed as a table during selection?

Currently I'm trying to allow selection from a column of data, but needing to pre-format as a table is making it impossible to make '--with-nth' and '--nth' work on the correct column, and extracting the column I want from the output is also proving difficult.

eg, input
```
ID,Code,Detail,Comment
1,en,English US,Default language
2,de,Gerrman,Secondary Display Language
3,cn,Chinese Simplified,Simplified Chinese Language (Requires UTF support)
```

fzf --header-lines 1 -d ',' --nth 2,3 --columns
```
1     en       English US           Default language
2     de       German,Secondary     Display Language
3     cn       Chinese Simplified   Simplified Chinese Language (Requires UTF support)
ID    Code     Detail               Comment
```

output after making a selection.
```
2,de,Gerrman,Secondary Display Language
````

Thanks,  I hope this makes sense.   I'm really struggling with FZF and selecting the correct column in data which contains spaces.

**Comment by junegunn** [comment]
fzf tries hard to follow the Unix philosophy so it doesn't implement something that can be delegated to other programs. If we are to implement column formatting inside fzf, we can't avoid thinking about a bunch of minor options such as the number of spaces between the columns, column alignment, colors, etc. But those concerns should be addressed by specialized formatting programs, not fzf.

> needing to pre-format as a table is making it impossible to make '--with-nth' and '--nth' work on the correct column

You can easily work around the problem by using non-breaking spaces. For example,

```sh
find * | column -s/ -t | sed $'s/ \([^ ]\)/\u00a0\\1/g' |
  fzf -d $'\u00a0' --nth 1 |
  awk -F $'\u00a0' '{print $2}'
```

**Comment by curiositycasualty** [comment]
So the (code) infrastructure for accepting and exec'ing a shell command input as an argument is already there in the form of `--preview`.

Accepting a similar `--format` or `--interface` argument whereby a user could supply `column -t -s=','` which would be applied to fzf input solely for the purposes of display during use of fzf (and not applying to its output) would seem to satisfy both unix methodology and your desire to not want to need to manage column-related code minutiae?

**Comment by LuRsT** [comment]
@curiositycasualty, I'm conflicted about that feature request because that would be adding some complexity into `fzf` where it's not really necessary, it shouldn't be `fzf`s concern to format the input.

But I see the point of displaying things differently during the execution, right now I've been simply formatting the input and then using `awk` to only do what I want during execution, which covers 99% of the scenarios.

I'd say no simply because there's a lot of complexity involved that can be avoided with the use of pipes for some edge cases.
